src/components/PublishedBooks/AddPublishedBooksModal.js

```diff
@@ -1,45 +1,125 @@
 import { FaRegTimesCircle } from "react-icons/fa";
+import { useSearchBook } from "../shared/searchBook";
+import { useState } from "react";
+import { GetISBN13 } from '../../functions/AddBook';
+import { AddBookToFirestore } from "../../functions/AddBook";
+import "../../styles/bookSearch.css";
+import {db,auth} from "../../firebase.js";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { fetchAuthorName } from "../../functions/fetchAuthorName.js";
+import {AddBookToAuthorPublishedBooks} from "../../functions/AddBookToAuthorPublishedBooks.js";
+import { AddNewPublishedBook } from "./AddNewPublishedBook.js";
+import { Link } from "react-router-dom";
 
-const AddPublishedBooksModal = () =>{
-    return (
-        <>
-          <div className="flex items-center justify-center fixed top-0 left-0 right-0 bottom-0 bg-gradient-to-b from-maroon z-10">
-        <div className="overlay-inner rounded-lg bg-lightcolor w-700 h-550 p-6 relative box-border overflow-hidden ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600">
-          <button className="close">
+
+
+const AddPublishedBooksModal = ({ show, onClose, user }) => {
+  const [searchBook, setSearchBook] = useState(null);
+  // const [user] = useAuthState(auth);
+  const searchResults = useSearchBook(searchBook);
+  const SearchBook = (e) => {
+    e.preventDefault();
+    setSearchBook(document.getElementById("book").value);
+  };
+  const handleBookClick = async(book) => {
+    console.log("Book Clicked: ", book);
+    AddBookToFirestore(book, null);
+    console.log("Book added to Firestore.");
+    const AuthorName = await fetchAuthorName(user.uid);
+    console.log("Existing author name: ",AuthorName);
+    const AuthorsFromBook = book.volumeInfo?.authors;
+
+    if (AuthorName && AuthorsFromBook.includes(AuthorName)) {
+      console.log("AuthorName exists in AuthorsFromBook");
+        await AddBookToAuthorPublishedBooks(user.uid, GetISBN13(book));
+        console.log("Book added to Author's published books.");
+      
+  } else {
+      console.log("AuthorName does not exist in AuthorsFromBook");
+  }
+  // onClose();
+};
+  return (
+    <>
+      <div className="flex items-center justify-center fixed top-0 left-0 right-0 bottom-0  bg-gradient-to-b from-maroon z-10">
+        <div className="overlay-inner rounded-lg bg-lightcolor w-700 h-550 p-6 relative box-border ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600">
```

Comment:
tailwind width and height only go up to w-96 and h-96 unless you've defined them in the tailwind.config as 

`module.exports = {
  theme: {
    extend: {
      width: {
        '500': '500px',
      },
    },
  },`

Reply:
I am actually not sure why that happened. I copied from an already existing code in our project. But I'll fix it.

Reply:
I removed those lines. The sizing is coming from a CSS file.